### PR TITLE
EditDataSources: Add EditDataSourceActions to EditDataSourcePages

### LINKED ISSUE #58311: Update Header of Data source Edit connection page
The data source "Edit" page should look like this:
https://www.figma.com/file/2L3Fqq7D2iUQWbS1B1uqTJ/Connections?node-id=3508%3A89439

Which means that there should be the following buttons in the right side of the header:
- build a dashboard
- explore
- delete
- save changes

### PR DESCRIPTION
**Pull request by mikkancso**
**What is this feature?**

Adding action buttons in the header of EditDataSourcePage at both occurrences, namely under Connections and Admin.
[Screencast from 2023-03-09 09-00-36.webm](https://user-images.githubusercontent.com/13637610/223963567-0e0ac12d-fc45-480c-a3aa-ae85ef9f3ec9.webm)

UPDATE:
[Topnav OFF]
Do not show buttons in the header. Leave buttons as they are in the footer.

Screenshot from 2023-04-05 11-40-57
Screenshot from 2023-04-05 11-40-50

[Topnav ON]
Header:
- build a dashboard
- explore
- delete

Screenshot from 2023-04-05 11-38-54

Footer:
- explore
- save & test

That also means we want to remove the Back button.

Screenshot from 2023-04-05 11-39-15

**Which issue(s) does this PR fix?**:

Fixes #58311

### CODE REVIEW

**Review by mckn — APPROVED** [review]
Code looks good! I have one question. If the topnav feature toggle is disabled I get the following apperance. Is it by design?

Screenshot 2023-03-15 at 11 20 33

**Comment by mikkancso** [comment]
If topnav is turned off then we put action buttons below the title, and that's by design - in general. For this particular case, I don't know if this is what we want. @teddyba, could you please take a look at the picture above?

**Comment by mikkancso** [comment]
I'm sorry folks, that I left this PR hanging for so long, I had many other duties.

We talked about this with @teddyba, and we came to the following proposal:

[Topnav OFF]
Do not show buttons in the header. Leave buttons as they are in the footer.

[Topnav ON]
Header: 
- build a dashboard
- explore
- delete
Footer:
- explore
- save & test

That also means we want to remove the Back button. We hope users can use the new navigation to navigate back.
I'm going to implement this now.

File: public/app/features/datasources/components/EditDataSourceActions.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+const getStyles = (theme: GrafanaTheme2) => {
+  return {
+    button: css({
+      marginLeft: theme.spacing(2),
+    }),
+  };
+};
+
+interface ActionsProps {
```
Comment (leventebalogh) [inline public/app/features/datasources/components/EditDataSourceActions.tsx:22]: **Nit:** this could probably be just `Props`.

**Review by leventebalogh — APPROVED** [review]
LGTM 👍

**Comment by mikkancso** [comment]
Rebased the branch to re-trigger drone

**Pull request merged**